### ISSUE TITLE
Add Textures / Packs in Plugin

### ISSUE THREAD

**Issue by morshey77** [pmmp/PocketMine-MP#5246]
## Description
I had the idea to make texture packs easier to add, i.e. like a Block or an Item to be able to add it directly in the Manager (Factory for block / item). 

## Justification
Since PM4, blocks, items and others are starting to be much easier to add, people are starting to get into "mods" like on java but adding packs is more complex and can be done like on forge with mods, reviewed of course to match PMMP, but you get the idea. 

## Alternative methods
For me the best way to do it is not to touch PM, but to add Resource Pack classes to it to add textures directly to it so that you can add it directly here: "https://github.com/pmmp/PocketMine-MP/blob/stable/src/resourcepacks/ResourcePackManager.php". Only via a specific class, like the ZIP class but being able to define the manifest in code, being able to add the textures (files) directly in an array of the class and using the array keys to store the paths of the textures .

**Comment by natof** [comment]
Yeah, that would be cool, I'm a developer who likes making complex texture packs and being able to make plugins that look like java mods.

**Comment by dktapps** [comment]
I believe Dragonfly has an API like this, if anyone wants to borrow some ideas for implementing it.
